src/zmalloc.c

```diff
@@ -396,34 +396,50 @@ void zmadvise_dontneed(void *ptr) {
 #include <sys/stat.h>
 #include <fcntl.h>
 
-size_t zmalloc_get_rss(void) {
-    int page = sysconf(_SC_PAGESIZE);
-    size_t rss;
+/* Get the i'th field from "/proc/self/stats" note i is 1 based as appears in the 'proc' man page */
+int get_proc_stat_ll(int i, long long *res) {
     char buf[4096];
-    char filename[256];
-    int fd, count;
+    int fd, l;
     char *p, *x;
 
-    snprintf(filename,256,"/proc/%ld/stat",(long) getpid());
-    if ((fd = open(filename,O_RDONLY)) == -1) return 0;
-    if (read(fd,buf,4096) <= 0) {
+    if ((fd = open("/proc/self/stat",O_RDONLY)) == -1) return 0;
+    if ((l = read(fd,buf,sizeof(buf)-1)) <= 0) {
         close(fd);
         return 0;
     }
     close(fd);
+    buf[l] = '\0';
+    if (buf[l-1] == '\n') buf[l-1] = '\0';
+    if (i == -1) printf("%s\n", buf);
 
-    p = buf;
-    count = 23; /* RSS is the 24th field in /proc/<pid>/stat */
-    while(p && count--) {
-        p = strchr(p,' ');
+    /* Skip pid and process name (surrounded with parentheses) */
+    p = strchr(buf, ')');
```

Comment:
won't it be faster to use `strchar`, and call it repeatedly till we find the last one?
considering this field is at the beginning, and parsing RSS is done frequently.